pkg/kubectl/cmd/create_podsecuritypolicy_test.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	pspv1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	cmdtesting "k8s.io/kubernetes/pkg/kubectl/cmd/testing"
+)
+
+func TestCreatePSP(t *testing.T) {
+	pspName := "my-psp"
+
+	tf := cmdtesting.NewTestFactory()
+	tf.Namespace = "test"
+	tf.Client = &fake.RESTClient{}
+	tf.ClientConfigVal = defaultClientConfig()
+
+	tests := map[string]struct {
+		seLinux            string
+		runAsUser          string
+		supplementalGroups string
+		fsGroup            string
+		expectedOutput     *pspv1beta1.PodSecurityPolicy
+	}{
+		"empty param": {
+			seLinux:            "",
+			runAsUser:          "",
+			supplementalGroups: "",
+			fsGroup:            "",
+
+			expectedOutput: &pspv1beta1.PodSecurityPolicy{
```

Comment:
Consider extraction a function that will take name and 4 strategies and return PSP object. This will simplify code and tests.

Reply:
I think construct object in simplest way is a better solution.